__tests__/app.test.js

```diff
@@ -0,0 +1,45 @@
+const app = require("../app");
+const request = require("supertest");
+const db = require("../db/connection.js");
+const seed = require("../db/seeds/seed.js");
+const data = require("../db/data/test-data/index.js");
+
+beforeEach(() => {
+  return seed(data);
+});
+afterAll(() => db.end());
+
+describe("/api/whaaaaaaaa???", () => {
+  test("GET:404 receive 404 and relevant message if trying to access non existant endpoint", () => {
+    return request(app)
+      .get("/api/whaaaaaaaa???")
+      .expect(404)
+      .then((response) => {
+        expect(response.body.msg).toBe("Route not found");
+      });
+  });
+});
+
+describe("/api/topics", () => {
+  test("GET:200 sends an array of topics to the client", () => {
+    return request(app)
+      .get("/api/topics")
+      .expect(200)
+      .then((response) => {
+        console.log(response.body.topics);
+        expect(response.body.topics.length).toBe(3);
+        response.body.topics.forEach((topic) => {
+          expect(topic).toHaveProperty("slug", expect.any(String));
+          expect(topic).toHaveProperty("description", expect.any(String));
+        });
+      });
+  });
+  test("POST: should return 405 for invalid method", () => {
```

Comment:
By not making an endpoint that accepts a post would cover this. By only building an endpoint with get attached to it that is the only valid method. 

An advanced task of this repo would ask you to allow posting to this endpoint should you wish to make new topics for new articles.

Reply:
Gotcha, the task said to consider errors and this is the only one I could think of. Will remove it 👍